ddtrace/llmobs/_evaluators/ragas/faithfulness.py

```diff
@@ -1,7 +1,49 @@
+import json
 import math
-import time
+from typing import List
 from typing import Optional
 
+from ddtrace.internal.logger import get_logger
+from ddtrace.llmobs._constants import RAGAS_ML_APP_PREFIX
+
+
+logger = get_logger(__name__)
+
+RAGAS_DEPENDENCIES_PRESENT = False
+RAGAS = "ragas"
+
+try:
+    from ragas.llms import llm_factory
+    from ragas.llms.output_parser import RagasoutputParser
+    from ragas.metrics import faithfulness
+    from ragas.metrics.base import ensembler
+    from ragas.metrics.base import get_segmenter
+
+    from ddtrace.llmobs._evaluators.ragas.models import StatementFaithfulnessAnswers
+    from ddtrace.llmobs._evaluators.ragas.models import StatementsAnswers
+
+    RAGAS_DEPENDENCIES_PRESENT = True
+except ImportError as e:
+    logger.warning("RagasFaithfulnessEvaluator is disabled because Ragas requirements are not installed", exc_info=e)
+
+
+def _get_ml_app_for_ragas_trace(span_event: dict) -> str:
+    """
+    The `ml_app` spans generated from traces of ragas will be named as `ragas-<ml_app>`
+    or `ragas` if `ml_app` is not present in the span event.
+    """
+    ml_app_of_span_event = span_event.get("ml_app")
+    if not ml_app_of_span_event:
+        return RAGAS_ML_APP_PREFIX
+    return "{}-{}".format(RAGAS_ML_APP_PREFIX, ml_app_of_span_event)
+
+
+def _get_faithfulness_instance():
```

Comment:
This is to ensure the `faithfulness` instance used in ragas evaluator is updated with the latest ragas faithfulness instance AND has an non-null `llm`